src/main/scala/strawman/collection/immutable/ListSet.scala

```diff
@@ -124,5 +127,10 @@ object ListSet extends IterableFactory[ListSet] {
 
   def empty[A]: ListSet[A] = EmptyListSet.asInstanceOf[ListSet[A]]
 
+  def newBuilder[A](): Builder[A, ListSet[A]] =
+    new ImmutableBuilder[A, ListSet[A]](empty) {
+      def add(elem: A): this.type = { elems = elems + elem; this }
```

Comment:
That's not much better, and we want to be really careful about per-element overhead.  I was thinking about pushing it up into a `SetFactory` which extends `IterableFactory`.  Not sure if there will be performance consequences, though.